### PR TITLE
Agent: prepare to fix chat parity

### PR DESCRIPTION
**Pull request by mkondratek**
I continue Olaf's work started there: https://github.com/sourcegraph/cody/pull/2396.

The related PR in JetBrains: https://github.com/sourcegraph/jetbrains/pull/220.

## Test plan
....

### CODE REVIEW

File: agent/src/agent.ts [hunk 1]
```diff
@@ -472,11 +537,100 @@ export class Agent extends MessageHandler {
...
+            }
+            const panel = this.webPanels.panels.get(id)
+            if (!panel) {
+                throw new Error(`No panel with id ${id} found`)
+            }
+            if (panel.isMessageInProgress) {
+                throw new Error('Message is already in progress')
+            }
+
+            const abortController = new AbortController()
```
Comment (mkondratek) [inline agent/src/agent.ts:585]: here 😕

File: vscode/src/chat/chat-view/SimpleChatPanelProvider.ts [hunk 2]
```diff
@@ -909,6 +910,7 @@ export class SimpleChatPanelProvider implements vscode.Disposable, IChatPanelPro
         userInputContextFiles?: ContextFile[],
         addEnhancedContext = true
     ): Promise<void> {
+        console.log({ addEnhancedContext })
```
Comment (valerybugakov) [inline vscode/src/chat/chat-view/SimpleChatPanelProvider.ts:913]: Left over?

File: vscode/src/chat/chat-view/SimpleChatPanelProvider.ts [hunk 3]
```diff
@@ -530,6 +530,7 @@ export class SimpleChatPanelProvider implements vscode.Disposable, IChatPanelPro
         userContextFiles: ContextFile[],
         addEnhancedContext: boolean
     ): Promise<void> {
+        console.log({ addEnhancedContext })
```
Comment (valerybugakov) [inline vscode/src/chat/chat-view/SimpleChatPanelProvider.ts:533]: Left over?

File: agent/src/vscode-shim.ts [hunk 4]
```diff
@@ -96,91 +103,33 @@ export let clientInfo: ClientInfo | undefined
...
 
     return (
-        connectionConfig.accessToken !== newConfig.accessToken ||
-        connectionConfig.serverEndpoint !== newConfig.serverEndpoint
+        extensionConfiguration.accessToken !== newConfig.accessToken ||
+        extensionConfiguration.serverEndpoint !== newConfig.serverEndpoint
     )
 }
 
-const configuration: vscode.WorkspaceConfiguration = {
```
Comment (valerybugakov) [inline agent/src/vscode-shim.ts:120]: Double-checking: is this removal intentional?

Comment (mkondratek) [reply agent/src/vscode-shim.ts:120]: This change has been made by Olaf originally. 

From what I understand, the configuration is extracted to `AgentWorkspaceConfiguration` - seems reasonable and intentional.

**Review by olafurpg — APPROVED** [review]
Took a quick look at the diffs after my last commit. Very nice job tracking down that long tail of gnarly issues! Super nice to see the CI be green 🎉 

* Node 16 issues likely means we didn't have enough test coverage to catch those before
* Very nice fix for flaky e2e tests by replacing hardcoded timeouts 


LGTM 👍🏻 This seems like a relatively safe diff to merge since the non-agent changes are minimal. The changes in the agent might introduce some regressions for clients that don't handle the new webview errors so we should post in #wg-cody-agent to let people know about this PR.
